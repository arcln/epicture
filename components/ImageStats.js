import React from 'react';
import {
  StyleSheet,
  View,
  Text,
  Platform,
  TouchableOpacity,
} from 'react-native';
import IconButton from '../components/IconButton';

export default class ImageStats extends React.Component {
  onUpvote() {
    alert('upvote');
  }

  onDownvote() {
    alert('downvote');
  }

  render() {
    return (
      <View style={{flexDirection: 'row'}}>
        <TouchableOpacity onPress={this.onUpvote} style={styles.infoButton}>
          <IconButton
            size={this.props.size + this.platformSettings.arrowsSize}
            color={this.props.color}
            name={this.platformSettings.arrowUp}
          />
          <View style={styles.buttonLabel}>
            <Text style={{fontSize: this.props.size, color: this.props.color}}>{this.props.data.ups || 0}</Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity onPress={this.onDownvote} style={styles.infoButton}>
          <IconButton
            size={this.props.size + this.platformSettings.arrowsSize}
            color={this.props.color}
            name={this.platformSettings.arrowDown}
          />
          <View style={styles.buttonLabel}>
            <Text style={{fontSize: this.props.size, color: this.props.color}}>{this.props.data.downs || 0}</Text>
          </View>
        </TouchableOpacity>
        <View style={styles.infoButton} >
          <View style={{justifyContent: 'center'}}>
            <IconButton
              size={this.props.size  + this.platformSettings.eyeSize}
              color={this.props.color}
              name={Platform.OS === 'ios' ? 'ios-eye' : 'md-eye'}
            />
          </View>
          <View style={{justifyContent: 'center', paddingLeft: 6}}>
            <Text style={{fontSize: this.props.size, color: this.props.color}}>{this.props.data.views || 0}</Text>
          </View>
        </View>
      </View>
    );
  }

  platformSettings = Platform.OS === 'ios' ? {
    arrowUp: 'ios-arrow-round-up',
    arrowDown: 'ios-arrow-round-down',
    arrowsSize: 8,
    eyeSize: 4,
  } : {
    arrowUp: 'md-arrow-round-up',
    arrowDown: 'md-arrow-round-down',
    arrowsSize: 4,
    eyeSize: 4,
  }
}

const styles = StyleSheet.create({
  infoButton: {
    paddingLeft: 5,
    paddingRight: 5,
    flexDirection: 'row',
    backgroundColor: 'transparent',
  },
  buttonLabel: {
    justifyContent: 'center',
    paddingLeft: 5,
  }
});
