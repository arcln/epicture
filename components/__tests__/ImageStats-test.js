import 'react-native';
import React from 'react';
import ImageStats from '../ImageStats';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <ImageStats
      size={42}
      color='#eeffaa'
      data={{
        ups: 1200,
        downs: 200,
        views: 22222200,
      }}
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
