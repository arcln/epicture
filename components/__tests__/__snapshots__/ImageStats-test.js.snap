// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingLeft": 5,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
    <View
      style={
        Object {
          "justifyContent": "center",
          "paddingLeft": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#eeffaa",
            "fontSize": 42,
          }
        }
      >
        1200
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingLeft": 10,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
    <View
      style={
        Object {
          "justifyContent": "center",
          "paddingLeft": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#eeffaa",
            "fontSize": 42,
          }
        }
      >
        200
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingLeft": 10,
        "paddingRight": 5,
      }
    }
  >
    <View
      style={
        Object {
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "transparent",
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
    </View>
    <View
      style={
        Object {
          "justifyContent": "center",
          "paddingLeft": 6,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#eeffaa",
            "fontSize": 42,
          }
        }
      >
        22222200
      </Text>
    </View>
  </View>
</View>
`;
